integration_test/metadata/integration_metadata.go

```diff
@@ -121,13 +121,16 @@ type IntegrationMetadata struct {
 	ExpectedMetricsContainer `yaml:",inline"`
 }
 
-func UnmarshalAndValidate(data []byte, i interface{}) error {
+func UnmarshalAndValidate(fullPath string, data []byte, i interface{}) error {
 	data = bytes.ReplaceAll(data, []byte("\r\n"), []byte("\n"))
 
 	v := NewIntegrationMetadataValidator()
 	// Note: Unmarshaler does not protect when only the key being declared.
 	// https://github.com/goccy/go-yaml/issues/313
-	return yaml.UnmarshalWithOptions(data, i, yaml.DisallowUnknownField(), yaml.Validator(v))
+	if err := yaml.UnmarshalWithOptions(data, i, yaml.DisallowUnknownField(), yaml.DisallowDuplicateKey(), yaml.Validator(v)); err != nil {
+		return fmt.Errorf("%s%w", fullPath, err)
```

Comment:
Nit: can we unsquish the `%s` and the `%w`? Or will they be implicitly separated by a newline or something?

Reply:
No, this is intentional. The YAML parser [reports the position of the error](https://github.com/goccy/go-yaml/blob/master/internal/errors/error.go#L187) in the data as `[<line>:<column>]` at the start of the error. We are prepending the filename to that, so the position immediately follows the filename with no spaces.